### PR TITLE
Translate Project Using po4a

### PR DESCRIPTION
**Pull request by danyeaw**
This PR moves from trying to use weblate's Javascript translation CDN to generating more standard gettext PO files using po4a of the Markdown files.

Links are added to the top of the main page for two initial languages: Croatian and Hungarian
image

### CODE REVIEW

**Comment by danyeaw** [comment]
@amolenaar What do you think of this approach to use po4a to translate the markdown files? It is what https://gitlab.com/fdroid/fdroid-website/ and some other Jeykll open source projects are using.

**Comment by amolenaar** [comment]
It looks nice, recognisable ;) It looks better than having allthose `l10n` statements throughout the pages.

The capture blocks are a bit peculiar. It should be able to extract text from (x)html, right?

**Comment by danyeaw** [comment]
It doesn't seem like po4a can process html inside of markdown. I wrote a Python script that goes through the markdown files and filters out the html, liquid tags, code blocks, and other things and then that gets passed to po4a now to create po files out of. Does this look better?

**Comment by danyeaw** [comment]
This is just about ready for at least an initial version. It looks like there is a style missing on the translated blog posts which I still need to look in to.

File: _sass/_languages.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#header {
+}
+
+.language-header {
+  width: 100%;
+  background-color: #292825;
+
+  .language-navigation {
+    float: right;
```
Comment (amolenaar) [inline _sass/_languages.scss:9]: Styling-wise we should not use `float` anymore. I know it's still in used in 1 or 2 places in the website, but it's been made obsolete by flex boxes (which allow for better styling rules, also on mobile devices).

File: po/po4a.conf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+[po4a_alias:markdown] text opt:"--option markdown" opt:"--option yfm_keys=title" opt:"--addendum-charset=UTF-8" opt:"--localized-charset=UTF-8" opt:"--master-charset=UTF-8" opt:"--keep=0"
+
+[type: markdown] _includes/footer.md pot_in:_includes/footer.md.filt $lang:$lang/_includes/footer.md
+[type: markdown] _includes/content.md pot_in:_includes/content.md.filt $lang:$lang/_includes/content.md
+[type: markdown] _platforms/01-windows.md pot_in:_platforms/01-windows.md.filt $lang:$lang/_platforms/01-windows.md
+[type: markdown] _platforms/02-macos.md pot_in:_platforms/02-macos.md.filt $lang:$lang/_platforms/02-macos.md
+[type: markdown] _platforms/03-linux.md pot_in:_platforms/03-linux.md.filt $lang:$lang/_platforms/03-linux.md
+[type: markdown] _platforms/04-python.md pot_in:_platforms/04-python.md.filt $lang:$lang/_platforms/04-python.md
+[type: markdown] _posts/2020-11-27-simple-services-in-python.md pot_in:_posts/2020-11-27-simple-services-in-python.md.filt $lang:$lang/_posts/2020-11-27-simple-services-in-python.md
```
Comment (amolenaar) [inline po/po4a.conf:14]: I do not really feel comfortable translating the blog posts as well. I can not understand the translations, but they have my name above it :thinking:.

File: _posts/2022-01-16-gaphor-2.8-highlights.md [hunk 3]
```diff
@@ -69,7 +70,7 @@ something is implicitly removed from the model.
 
 ## AppImage and Homebrew
 
-The Gaphor [AppImage](/download.html#appimage) has been built against Ubuntu
+The Gaphor [AppImage](% link download.md %}#appimage) has been built against Ubuntu
```
Comment (amolenaar) [inline _posts/2022-01-16-gaphor-2.8-highlights.md:73]: missing `{` before `% link`?

File: po/po4a.conf (hunk 2 above)
Comment (danyeaw) [reply po/po4a.conf:14]: Ya that makes sense, I removed the blog posts from being translated. :+1:

**Comment by danyeaw** [comment]
> The language selection is only available on the front page. Shouldn't it be available on the other pages as well? Maybe it can be moved to a dropdown next to the github icon?

Updated to be on each page :+1:

File: _sass/_languages.scss (hunk 1 above)
Comment (danyeaw) [reply _sass/_languages.scss:9]: I was just borrowing from the css for the header. I agree that using the flex boxes would be better, but should we update it in a follow-on PR?

**Comment by danyeaw** [comment]
@amolenaar Any other changes required before we merge this?

**Comment by amolenaar** [comment]
@danyeaw Looks good. I made a few more tweaks (on `main`). Now the menu is not jumping between the main and sub pages.

**Comment by danyeaw** [comment]
@amolenaar Thanks! 🙌

**Pull request merged**